source/Mission.cpp

```diff
@@ -1248,19 +1268,30 @@ Mission Mission::Instantiate(const PlayerInfo &player, const shared_ptr<Ship> &b
 	while(!destinations.empty())
 	{
 		// Find the closest destination to this location.
-		DistanceMap distance(path);
+		DistanceMap distance(path, routeOptions.wormholeStrategy, routeOptions.requiresJumpDrive);
 		auto it = destinations.begin();
 		auto bestIt = it;
+		int bestDays = distance.Days(*bestIt);
+		if(bestDays < 0)
+			bestDays = numeric_limits<int>::max();
 		for(++it; it != destinations.end(); ++it)
-			if(distance.Days(*it) < distance.Days(*bestIt))
+		{
+			int days = distance.Days(*it);
+			if(days >= 0 && days < bestDays)
+			{
 				bestIt = it;
+				bestDays = days;
+			}
+		}
 
 		path = *bestIt;
-		jumps += distance.Days(*bestIt);
+		// If currently unreachable, this system does not add to the deadline.
+		jumps += max(0, distance.Days(*bestIt));
```

Comment:
```suggestion
		jumps += max(0, bestDays);
```

? Although you'd have to refactor the code above a bit so that you don't add the highest int value to `jumps`.

Reply:
or that lol